client/src/components/App/Content/InvestigationForm/TabManagement/PersonalInfoTab/PersonalInfoTabValidationSchema.ts

```diff
@@ -41,8 +42,11 @@ const schema = yup.object().shape({
             schema.nullable().required(requiredText) :
             schema.nullable()
     }),
-    [PersonalInfoDataContextFields.EDUCATION_CLASS_NUMBER]: yup.number().transform((value: any) => 
-        (Boolean(value)) ? value : null).nullable().max(50, maxClassNumberError)
+    [PersonalInfoDataContextFields.EDUCATION_CLASS_NUMBER]: yup.number().when(['role', 'relevantOccupation'],  (role: number, relevantOccupation :any, schema: any) => {
```

Comment:
You have to clean the field value, because the value is saved although the change of occupation

Reply:
its not part of the bug, you can take a look at bug 1587